### PR TITLE
1887 cache django queries

### PR DESCRIPTION
**Pull request by jithendralal**
Used caching for: 

- Registry - forms 
- RegistryForm - section_models
- Section - cde_models
- Survey - client_rep

### CODE REVIEW

File: rdrf/rdrf/helpers/cache_utils.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+env = EnvConfig()
+
+
+def use_cache(function):
+    def wrapper(*args, **kwargs):
+
+        if settings.CACHE_DISABLED or env.get("CACHE_DISABLED", False):
+            return function(*args, **kwargs)
+
+        key = f"{function.__name__}"
```
Comment (id2359) [inline rdrf/rdrf/helpers/cache_utils.py:16]: this is a static key so won't work  when arguments are different

Comment (id2359) [reply rdrf/rdrf/helpers/cache_utils.py:16]: best to remove this if not being used

File: rdrf/rdrf/helpers/cache_utils.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            return query_cache.get(key)
+        else:
+            result = method(*args, **kwargs)
+            logger.info(f"Caching: {key}")
+            query_cache.set(key, result)
+            return result
+    return wrapper
+
+
+def get_cache(key=None):
```
Comment (id2359) [inline rdrf/rdrf/helpers/cache_utils.py:47]: remove unused code

File: rdrf/rdrf/helpers/cache_utils.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        obj = args[0]
+        key = f"{method.__qualname__}_id{str(obj.pk)}"
+        query_cache = caches["queries"]
+        if key in query_cache:
+            logger.info(f"Fetching from Cache: {key}")
+            return query_cache.get(key)
+        else:
+            result = method(*args, **kwargs)
+            logger.info(f"Caching: {key}")
```
Comment (id2359) [inline rdrf/rdrf/helpers/cache_utils.py:41]: remove info logging

File: rdrf/rdrf/models/proms/models.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 from rdrf.models.definition.models import CommonDataElement
 from rdrf.models.definition.models import ContextFormGroup
 from rdrf.helpers.utils import generate_token
+from rdrf.helpers.cache_utils import use_object_cache
```
Comment (id2359) [inline rdrf/rdrf/models/proms/models.py:15]: use_query_cache as a name ?

File: rdrf/rdrf/services/rest/views/proms_api.py [hunk 5]
```diff
@@ -34,6 +35,16 @@ def multicde(cde_model):
     return datatype == "range" and cde_model.allow_multiple
 
 
+@use_cache
+def get_registry(code=None):
+    return Registry.objects.get(code=code)
```
Comment (id2359) [inline rdrf/rdrf/services/rest/views/proms_api.py:40]: Best to remove use_cache and calls to it for now

**Review by id2359 — CHANGES_REQUESTED** [review]

File: rdrf/rdrf/helpers/cache_utils.py (hunk 1 above)
Comment (jithendralal) [reply rdrf/rdrf/helpers/cache_utils.py:16]: removed

File: rdrf/rdrf/helpers/cache_utils.py (hunk 2 above)
Comment (jithendralal) [reply rdrf/rdrf/helpers/cache_utils.py:47]: removed

File: rdrf/rdrf/helpers/cache_utils.py (hunk 3 above)
Comment (jithendralal) [reply rdrf/rdrf/helpers/cache_utils.py:41]: logging removed

File: rdrf/rdrf/models/proms/models.py (hunk 4 above)
Comment (jithendralal) [reply rdrf/rdrf/models/proms/models.py:15]: renamed

**Review by id2359 — APPROVED** [review]

**Pull request merged**